python/lsst/pipe/tasks/extended_psf.py

```diff
@@ -239,3 +245,250 @@ def readFits(cls, filename):
+        return stats_control, stats_flags
+
+    def run(self, bss_ref_list, region_name=None):
+        """Read input bright star stamps and stack them together.
+
+        The stacking is done iteratively over smaller areas of the final model
+        image to allow for a great number of bright star stamps to be used.
+
+        Parameters
+        ----------
+        bss_ref_list : `list` of
+                `lsst.daf.butler._deferredDatasetHandle.DeferredDatasetHandle`
+            List of available bright star stamps data references.
+        region_name : `str`, optional
+            Name of the focal plane region, if applicable. Only used for
+            logging purposes, when running over multiple such regions
+            (typically from `MeasureExtendedPsfTask`)
+        """
+        log_message = f'Building extended PSF from stamps extracted from {len(bss_ref_list)} detector images '
+        if region_name:
+            log_message += f'for region "{region_name}".'
+        self.log.info(log_message)
+        # read in example set of full stamps
+        for bss_ref in bss_ref_list:
+            if not isinstance(bss_ref, DeferredDatasetHandle):
+                if not bss_ref.datasetExists("brightStarStamps"):
+                    self.log.warn("Could not find %s %s; skipping it", "brightStarStamps", bss_ref.dataId)
+                    bss_ref_list.remove(bss_ref)
+                    continue
+            bss = bss_ref.get(datasetType="brightStarStamps", immediate=True)
+            break
+        example_stamp = bss[0].stamp_im
+        # create model image
+        ext_psf = afwImage.MaskedImageF(example_stamp.getBBox())
+        # divide model image into smaller subregions
+        subregion_size = Extent2I(*self.config.subregion_size)
+        sub_bboxes = AssembleCoaddTask._subBBoxIter(ext_psf.getBBox(), subregion_size)
+        # compute approximate number of subregions
+        n_subregions = int(ext_psf.getDimensions()[0]/subregion_size[0] + 1)*int(
+            ext_psf.getDimensions()[1]/subregion_size[1] + 1)
+        self.log.info(f"Stacking will performed iteratively over approximately {n_subregions} "
+                      "smaller areas of the final model image.")
+        # set up stacking statistic
+        stats_control, stats_flags = self._set_up_stacking(example_stamp)
+        # perform stacking
+        for jbbox, bbox in enumerate(sub_bboxes):
+            all_stars = None
+            for bss_ref in bss_ref_list:
+                if not isinstance(bss_ref, DeferredDatasetHandle):
+                    if not bss_ref.datasetExists("brightStarStamps"):
+                        self.log.warn("Could not find %s %s; skipping it", "brightStarStamps", bss_ref.dataId)
+                        bss_ref_list.remove(bss_ref)
+                        continue
+                read_stars = bss_ref.get(datasetType="brightStarStamps", parameters={'bbox': bbox})
```

Comment:
As this is the second time you're checking each entry in `bss_ref_list` to check for `bss_ref`s that are valid `DeferredDatasetHandle` types and that also contain `brightStarStamps` dataset types, perhaps this functionality should be stripped out and placed somewhere higher up in the run task (before L342) to check and remove invalid entries at the outset. That way, you can raise an exception and quit early if no valid entries are found, and reduce the complexity of using `bss_ref_list` elsewhere in the code.